lib/jobsub_api.py

```diff
@@ -0,0 +1,62 @@
+from typing import List, Optional, Generator
+import re
+import sys
+import contextlib
+from io import StringIO
+from mains import jobsub_submit_main, jobsub_fetchlog_main, jobsub_cmd_main
+
+
+@contextlib.contextmanager
+def output_saver(should_i: bool) -> Generator[StringIO, bool, StringIO]:
+    """
+    context manager that optionally puts sys.stdio and sys.stderr into
+    a StringIO() so you can look at them...
+    """
+
+    output = StringIO()
+    if should_i:
+        # save initial stdout, stderr
+        save_out = sys.stdout
+        save_err = sys.stderr
+        # point them at our StringIO
+        sys.stdout = output
+        sys.stderr = output
+    yield output
+    if should_i:
+        # put them back
+        sys.stderr = save_err
+        sys.stdout = save_out
+    return output
+
+
+# so clients can easily parse the result strings
+jobsub_submit_re = re.compile(r"Use job id (?P<jobid>[0-9.]+\@[^ ]+) to retrieve")
+
+jobsub_q_re = re.compile(
```

Comment:
Same as above, but a bit trickier to do "right." Should we add the attributes to the existing `Job` class and return a list of those, or create a new `JobStatus` class specific for this? Simpler API (with some added internal complexity) or easier?

Reply:
I'd be in favor of adding the attribute to the Job class, and returning a list of those. We wouldn't even have to make up a new enumeration in that case, as the HTCondor python bindings provide those:

https://htcondor.readthedocs.io/en/latest/apis/python-bindings/api/htcondor.html#htcondor.JobStatus

Reply:
@retzkek - given that we plan on addressing these in #609, are you OK with us moving ahead with merging this after my re-review?